runtime/lua/vim/lsp.lua

```diff
@@ -1089,6 +1090,12 @@ function lsp.start_client(config)
         return true
       end
       if vim.tbl_get(client.server_capabilities, unpack(required_capability)) then
+        if method == ms.textDocument_rangesFormatting then
+          local range_formatting_provider =
+            client.server_capabilities.documentRangeFormattingProvider
+          return type(range_formatting_provider) == 'table'
+            and range_formatting_provider.rangesSupport
```

Comment:
I don't think this is the right pattern to deal with nested capabilities. Instead you should use a table as the value here:
https://github.com/neovim/neovim/blob/bebaf54d68ed3ebbe35003db2eeeedc59d7b56f3/runtime/lua/vim/lsp.lua#L60

Here's an example for `prepareRename`.
https://github.com/neovim/neovim/blob/cd3399f6bd99494c5805ee2560e515ec6c782e59/runtime/lua/vim/lsp.lua#L52

Reply:
Ohh right, I for some reason thought `vim.tbl_get` would take adjacent keys instead of nested ones.